pkgs/development/tools/godot/mono/default.nix

```diff
@@ -0,0 +1,94 @@
+{ godot
+, lib
+, callPackage
+, mkNugetDeps
+, mkNugetSource
+, mono
+, dotnet-sdk
+, writeText
+}:
+
+godot.overrideAttrs (self: base: {
+  pname = "godot-mono";
+
+  passthru = let bp = base.passthru; in bp // {
+    sconsFlagSet = bp.sconsFlagSet // {
+      module_mono_enabled = true;
+      mono_prefix = mono;
+    };
+
+    buildDescription = "mono build";
+
+    glue = callPackage ./glue.nix {};
+
+    make-deps = callPackage ./make-deps.nix {};
+  };
+
+  nativeBuildInputs = base.nativeBuildInputs ++ [ mono dotnet-sdk ];
+
+  nugetDeps = mkNugetDeps { name = "deps"; nugetDeps = import ./deps.nix; };
+
+  nugetSource =
+    mkNugetSource {
+      name = "${self.pname}-nuget-source";
+      description = "A Nuget source with dependencies for ${self.pname}";
+      deps = [ self.nugetDeps ];
+    };
+
+  nugetConfig = writeText "NuGet.Config" ''
+    <?xml version="1.0" encoding="utf-8"?>
+    <configuration>
+      <packageSources>
+        <add key="${self.pname}-deps" value="${self.nugetSource}/lib" />
+      </packageSources>
+    </configuration>
+  '';
```

Comment:
This should be in a let in and not be used via a bash variable.

Reply:
What do you mean?  I don't see a bash variable in there.

Reply:
Everything that mkDerivation does not understand and is not used as a nix variable is passed as a bash variable into the builder. Because of that variables like this should be in a let in.

Reply:
Oh i see, you mean `nugetConfig` itself should be passed into `configureNugetStep` as `${nugetConfig}`, rather than exposing it as a bash variable and using it as `$nugetConfig`.  Why?  If there's a good reason for this, I can move it into passthru, and rewrite `configureNugetStep` to do:
```sh
ln -s "${self.passthru.nugetConfig}" ~/.nuget/NuGet/NuGet.Config
```
But I'm not sure what this buys us.